packages/snap/src/dialogs/utxos.tsx

```diff
@@ -0,0 +1,96 @@
+import { constants as libConstants, numberUtils } from '@hathor/wallet-lib';
+import { Bold, Box, Container, Divider, Heading, Section, Text } from '@metamask/snaps-sdk/jsx';
+
+const renderTokenFilterParam = (token) => {
+  if (!token || token === libConstants.NATIVE_TOKEN_UID) {
+    return libConstants.DEFAULT_NATIVE_TOKEN_CONFIG.symbol;
+  }
+
+  return token;
+}
+
+// We show the symbol only if it's HTR in the list
+const renderTokenSymbol = (token) => {
+  if (!token || token === libConstants.NATIVE_TOKEN_UID) {
+    return libConstants.DEFAULT_NATIVE_TOKEN_CONFIG.symbol;
+  }
+
+  return '';
+}
+
+const renderAmountSummary = (data, params) => {
+  if (params.authorities) {
+    let authority = '';
+    if (params.authorities === 1) {
```

Comment:
suggestion(non-blocking): Use the lib constants for those authority values
```suggestion
    if (params.authorities === libConstants.TOKEN_MINT_MASK) {
```